src/Dfc.CourseDirectory.WebV2/Features/BulkUpload/BulkUploadController.cs

```diff
@@ -48,5 +48,14 @@ public IActionResult Regions()
         [HttpGet("/BulkUpload/PublishYourFile")]
         public async Task<IActionResult> CoursesPublishFile(ProviderContext providerContext) =>
             await _mediator.SendAndMapResponse(new CoursesPublishFile.Query{ProviderId = providerContext.ProviderInfo.ProviderId}, vm => View(vm));
+
+        [HttpGet("publishing-your-file")]
+        public IActionResult CoursesPublishingFile(int numberOfCourses)
```

Comment:
Could pass this over `TempData` instead of by query param

Reply:
You could, but if you refresh the page it'll vanish, it's also not RESTful

Reply:
Does RESTful as a concept work with web pages/websites? (I'm not sure it does.) Even if this concept was to be applied, for it to be truly RESTful, the URL would need to contain the ID of the resource that was being retrieved, in this case the bulk upload ID.

Reply:
@jameslawesynergy - seeing as I brought up REST :smile: ...

I like REST as a stretch goal because I think applications that attempt to follow the principles (particularly statelessness) are better systems. We're not going to get all the way because of the provider state in the session. Bulk upload has no separation of uploads currently so there isn't an id for it.

Web systems can be RESTful as described in these articles:
* https://en.wikipedia.org/wiki/Representational_state_transfer
* https://www.codecademy.com/articles/what-is-rest

People seem to mix up REST vs RPC with the JSON vs SOAP thing which is actually tangential.

----

Anyway, in my judgement the query param is a better fit here, and is not big departure from how the system was before so I'm inclined to leave it as unless anyone feels really strongly. Happy to jump on a call to discuss further if you like.